Respuestas Gastón Paris

gastonparis opened the pull request:
I'm sending the answers for dota notebook

gastonparis committed:
Respuestas Gastón Paris

metdinov commented:
Good job! I will improve the wording on exercise 9, thanks for the heads up. Your interpretation was correct, we wanted to find the _most frequently_ picked heroes for each side.

A couple of mentions:
- In exercise 7 you are plotting `result_df.sum(axis=1)` which is not the number of wins per team
- When you index using a boolean column (such as `df['radiant_win']`) keep in mind that

```python
>>> df.loc[df['radiant_win'] == True] == df.loc[df['radiant_win']]
```

- The calls to `.to_list()` could be very expensive if the dataset were much bigger. Also, as a general rule, avoid nested for comprehensions as they can become hard to understand.

gastonparis commented:
Thank you for your comments.

- I fixed exercise 7 by `result_df['total_wins'].plot(kind='bar')` 

- I replaced `df.loc[df['radiant_win'] == True]` to `df.loc[df['radiant_win'] ]` and, more interesting for me, replaced `df.loc[df['radiant_win'] == False]` to `df.loc[~df['radiant_win'] ]`
I didn't known about negate symbol ~

- I replaced all `.tolist()` calls by `.values`. I wasn't aware about the performance issues of .tolist() calls. 
Finally, I did my code more explicit by replacing list comprehensions:
```
wins = pd.Series([hero for heroes in df.loc[df['radiant_win'] == team[1]][team[0]+'_picks'].values 
                                   for hero in heroes], name='total_wins').value_counts()
```
by

```
wins = []
    for heroes in df.loc[df['radiant_win'] == team[1]][team[0]+'_picks'].values:
        for hero in heroes:
            wins.append(hero)
    wins = pd.Series(wins, name = 'total_wins').value_counts()
```
